### PR TITLE
feat(emx2): JSON api for reports

### PR DESCRIPTION
**Pull request by harmbrugge**
Fixes: [#4360](https://github.com/molgenis/molgenis-emx2/issues/4360)

What are the main changes you did:
- Added a JSON api mapping for reports
- Added a json download button to the reports view

how to test:
- Go to the pet store schema
- Click on reports, and select the first one
- Click on the download as json button
- See the json

todo:
- [x] support special case where the query returns 1 json objects
- [x] updated docs to explain us of this json
- [ ] add test for the single json blob

### CODE REVIEW

File: backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/JsonApi.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      Map<String, Object> reportObject = reportList.get(Integer.parseInt(reportId.trim()));
+      String sql = (String) reportObject.get("sql");
+      String name = (String) reportObject.get("name");
+      List<Row> rows = schema.retrieveSql(sql, parameters);
+
+      List<Map<String, Object>> result = new ArrayList<>();
+      for (Row row : rows) {
+        result.add(row.getValueMap());
+      }
+      jsonResponse.put(name, result);
```
Comment (mswertz) [inline backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/JsonApi.java:49]: this is nice, but ... we could make SQL that produces JSON itself. So I would propose we adapt this such that if the row has only one value, and that value is of type JSONB that we return that value? Could you try to make testcase report that produces JSON to test this? Then we could amend the docs to make that clear

**Review by mswertz — CHANGES_REQUESTED** [review]
very nice, puts a smile on my face, but one bonus request see comments

File: backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/JsonApi.java (hunk 1 above)
Comment (harmbrugge) [reply backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/JsonApi.java]: The reports indeed break if you use jsonb I added this.

**Comment by mswertz** [comment]
sorry, broke your PR

**Pull request merged**